.github/workflows/merge_workflow.yml

```diff
@@ -0,0 +1,103 @@
+name: CI/CD
+
+on:
+  push:
+    branches: [main]
+    tags: ['*']
+  pull_request:
+    branches: [main]
+
+env:
+  SCALA_VERSION: 2.13.8
+  JAVA_VERSION: 11
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
Using a matrix here feels a bit awkward since you need custom logic for each. Might be more clear to have two separate jobs that run in parallel.

Reply:
Good point, I split into separate jobs.